modules/apps/transfer/types/transfer_authorization_test.go

```diff
@@ -1,132 +1,161 @@
-	fromAddr       = sdk.AccAddress("_____from _____")
-	toAddr         = sdk.AccAddress("_______to________")
-	timeoutHeight  = clienttypes.NewHeight(0, 10)
-)
+func (suite *TypesTestSuite) TestTransferAuthorizationAccept() {
+	var (
+		msgTransfer   types.MsgTransfer
+		transferAuthz types.TransferAuthorization
+	)
 
-func TestTransferAuthorization(t *testing.T) {
-	app := simapp.Setup(t, false)
-	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
-	allocation := types.Allocation{
-		SourcePort:    sourcePort,
-		SourceChannel: sourceChannel,
-		SpendLimit:    coins1000,
-		AllowList:     []string{toAddr.String()},
-	}
-	authorization := types.NewTransferAuthorization(allocation)
-
-	t.Log("verify authorization returns valid method name")
-	require.Equal(t, authorization.MsgTypeURL(), "/ibc.applications.transfer.v1.MsgTransfer")
-	require.NoError(t, authorization.ValidateBasic())
-	transfer := types.NewMsgTransfer(sourcePort, sourceChannel, coin1000, fromAddr.String(), toAddr.String(), timeoutHeight, 0, "")
-	require.NoError(t, authorization.ValidateBasic())
-
-	t.Log("verify updated authorization returns nil")
-	resp, err := authorization.Accept(ctx, transfer)
-	require.NoError(t, err)
-	require.True(t, resp.Delete)
-	require.Nil(t, resp.Updated)
-
-	t.Log("verify updated authorization returns remaining spent limit")
-	authorization = types.NewTransferAuthorization(allocation)
-	require.Equal(t, authorization.MsgTypeURL(), "/ibc.applications.transfer.v1.MsgTransfer")
-	require.NoError(t, authorization.ValidateBasic())
-	transfer = types.NewMsgTransfer(sourcePort, sourceChannel, coin500, fromAddr.String(), toAddr.String(), timeoutHeight, 0, "")
-	require.NoError(t, authorization.ValidateBasic())
-	resp, err = authorization.Accept(ctx, transfer)
-	require.NoError(t, err)
-	require.False(t, resp.Delete)
-	require.NotNil(t, resp.Updated)
-
-	allocation = types.Allocation{
-		SourcePort:    sourcePort,
-		SourceChannel: sourceChannel,
-		SpendLimit:    coins500,
-		AllowList:     []string{toAddr.String()},
-	}
-	sendAuth := types.NewTransferAuthorization(allocation)
-	require.Equal(t, sendAuth.String(), resp.Updated.String())
-
-	t.Log("expect updated authorization nil after spending remaining amount")
-	resp, err = resp.Updated.Accept(ctx, transfer)
-	require.NoError(t, err)
-	require.True(t, resp.Delete)
-	require.Nil(t, resp.Updated)
-
-	t.Log("expect error when spend limit for specific port and channel is not set")
-	allocation = types.Allocation{
-		SourcePort:    sourcePort,
-		SourceChannel: sourceChannel,
-		SpendLimit:    coins1000,
-		AllowList:     []string{toAddr.String()},
-	}
-	authorization = types.NewTransferAuthorization(allocation)
-	transfer = types.NewMsgTransfer(sourcePort2, sourceChannel2, coin500, fromAddr.String(), toAddr.String(), timeoutHeight, 0, "")
-	_, err = authorization.Accept(ctx, transfer)
-	require.Error(t, err)
+	testCases := []struct {
+		name      string
+		malleate  func()
```

Comment:
[nit] for the malleate function, we use `func()` in almost all our tests I think, but what do you think about limiting the scope of what we can change and removing the test suite level variables 

```golang
var (
		msgTransfer   types.MsgTransfer
		transferAuthz types.TransferAuthorization
	)
```

altogether?

Something like

```golang
malleate func(msgTransfer types.MsgTransfer, transferAuthz types.TransferAuthorization)
```

I think this makes the intent far more clear than when no arguments are supplied and eliminates the potential for one test to accidentally effect another.